internal compiler errors using go1.7.6

bazel 0.5.4, rules_go 473417ec48310325e1fcb1c154621a83197a17fe

I updated the rules_go reference on one the older kubernetes release branches (release-1.6) to support newer versions of bazel, and now when trying to build with go1.7.6, I'm getting weird internal compiler errors:

```console
$ bazel build pkg/credentialprovider/aws/...
INFO: Analysed 4 targets (18 packages loaded).              
INFO: Found 4 targets...                                    
ERROR: [home]/code/kubernetes/src/k8s.io/kubernetes/pkg/credentialprovider/aws/BUILD:25:1: GoCompile pkg/credentialprovider/aws/~lib~/go_default_test.o failed (Exit 1): compile failed: er
ror executing command                                       
  (cd [home]/.cache/bazel/_bazel_jgrafton/75e5737b0f848bf1b7c3ab3303c8bf72/execroot/__main__ && \
  exec env - \                                              
    GOARCH=amd64 \                                          
    GOOS=linux \                                            
    GOROOT=[home]/.cache/bazel/_bazel_jgrafton/75e5737b0f848bf1b7c3ab3303c8bf72/external/go1_7_6_linux_amd64 \
  bazel-out/host/bin/external/io_bazel_rules_go/go/tools/builders/compile external/go1_7_6_linux_amd64/bin/go -src pkg/credentialprovider/aws/aws_credentials.go -src pkg/credentialprovider/aws/aws_credentials_tes
t.go -dep k8s.io/kubernetes/pkg/credentialprovider -dep github.com/aws/aws-sdk-go/aws -dep github.com/aws/aws-sdk-go/service/ecr -dep k8s.io/kubernetes/pkg/credentialprovider -dep github.com/aws/aws-sdk-go/aws -d
ep github.com/aws/aws-sdk-go/aws/request -dep github.com/aws/aws-sdk-go/aws/session -dep github.com/aws/aws-sdk-go/service/ecr -dep github.com/golang/glog -o 'bazel-out/local-fastbuild/bin/pkg/credentialprovider/
aws/~lib~/go_default_test.o' -trimpath . -I . -I 'bazel-out/local-fastbuild/bin/pkg/credentialprovider/~lib~/' -I 'bazel-out/local-fastbuild/bin/vendor/~lib~/' -I 'bazel-out/local-fastbuild/bin/vendor/~lib~/' -I 
'bazel-out/local-fastbuild/bin/pkg/credentialprovider/~lib~/' -I 'bazel-out/local-fastbuild/bin/vendor/~lib~/' -I 'bazel-out/local-fastbuild/bin/vendor/~lib~/' -I 'bazel-out/local-fastbuild/bin/vendor/~lib~/' -I 
'bazel-out/local-fastbuild/bin/vendor/~lib~/' -I 'bazel-out/local-fastbuild/bin/vendor/~lib~/' --)
                                                            
Use --sandbox_debug to see verbose messages from the sandbox
pkg/credentialprovider/aws/aws_credentials.go:28: internal compiler error: importer: expected package tag, found tag = -28
goroutine 1 [running]:                                      
runtime/debug.Stack(0x0, 0x0, 0x0)                          
        /usr/local/go/src/runtime/debug/stack.go:24 +0x79   
cmd/compile/internal/gc.Fatalf(0x90c78e, 0x2e, 0xc42079a5d8, 0x1, 0x1)
        /usr/local/go/src/cmd/compile/internal/gc/subr.go:165 +0x248
cmd/compile/internal/gc.(*importer).pkg(0xc42049a000, 0x32) 
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:221 +0xa2a
cmd/compile/internal/gc.(*importer).fieldName(0xc42049a000, 0x0)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:603 +0x397
cmd/compile/internal/gc.(*importer).field(0xc42049a000, 0x5)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:545 +0x3d
cmd/compile/internal/gc.(*importer).fieldList(0xc42049a000, 0xffffffffffffff16, 0xc4208ddf40, 0x90)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:536 +0x8f
cmd/compile/internal/gc.(*importer).typ(0xc42049a000, 0xc4208ddc20)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:483 +0xc77
cmd/compile/internal/gc.(*importer).typ(0xc42049a000, 0xffffffffffffff12)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:407 +0x148
cmd/compile/internal/gc.(*importer).typ(0xc42049a000, 0xffffffffffffff14)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:487 +0xfde
cmd/compile/internal/gc.(*importer).typ(0xc42049a000, 0xc420952690)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:474 +0xbf4
cmd/compile/internal/gc.(*importer).field(0xc42049a000, 0xc420977710)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:546 +0x55
cmd/compile/internal/gc.(*importer).fieldList(0xc42049a000, 0xffffffffffffff16, 0xc4208dd450, 0x3f)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:536 +0x8f
cmd/compile/internal/gc.(*importer).typ(0xc42049a000, 0xc4208dd180)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:483 +0xc77
cmd/compile/internal/gc.(*importer).typ(0xc42049a000, 0x0)  
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:407 +0x148
cmd/compile/internal/gc.(*importer).obj(0xc42049a000, 0xfffffffffffffffd)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:282 +0x967
cmd/compile/internal/gc.Import(0xc42049e000)                
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:97 +0x3b2
cmd/compile/internal/gc.importfile(0xc420424800, 0x0, 0x0, 0x0)
        /usr/local/go/src/cmd/compile/internal/gc/main.go:843 +0x9b1
cmd/compile/internal/gc.(*parser).importdcl(0xc4204247e0)   
        /usr/local/go/src/cmd/compile/internal/gc/parser.go:341 +0x178
cmd/compile/internal/gc.(*parser).import_(0xc4204247e0)     
        /usr/local/go/src/cmd/compile/internal/gc/parser.go:300 +0x71
cmd/compile/internal/gc.(*parser).file(0xc4204247e0)        
        /usr/local/go/src/cmd/compile/internal/gc/parser.go:268 +0x5e
cmd/compile/internal/gc.parse_file(0xc420074240)            
        /usr/local/go/src/cmd/compile/internal/gc/parser.go:31 +0x54
cmd/compile/internal/gc.Main()                              
        /usr/local/go/src/cmd/compile/internal/gc/main.go:334 +0x14f4
cmd/compile/internal/amd64.Main()                           
        /usr/local/go/src/cmd/compile/internal/amd64/galign.go:93 +0x2fa
main.main()                                                 
        /usr/local/go/src/cmd/compile/main.go:33 +0x2a3     
```

The `BUILD` files are a bit of a mess in this branch (we have one `BUILD` file for all of `vendor/`), but updating to use multiple `BUILD` files doesn't seem to help.

I can reproduce on kubernetes master as well, with a slightly different error:
```console
$ git diff
diff --git a/build/root/WORKSPACE b/build/root/WORKSPACE
index f92e09dded..95e6e362c7 100644
--- a/build/root/WORKSPACE
+++ b/build/root/WORKSPACE
@@ -43,7 +43,7 @@ load("@io_bazel_rules_go//go:def.bzl", "go_repositories")
 load("@io_bazel_rules_docker//docker:docker.bzl", "docker_repositories", "docker_pull")
 
 go_repositories(
-    go_version = "1.8.3",
+    go_version = "1.7.6",
 )
 
 docker_repositories()

$ bazel build pkg/credentialprovider/aws/...
INFO: Analysed 4 targets (1 packages loaded).
INFO: Found 4 targets...
ERROR: [home]/code/kubernetes/src/k8s.io/kubernetes/pkg/credentialprovider/aws/BUILD:22:1: GoCompile pkg/credentialprovider/aws/~lib~/go_default_test.o failed (Exit 1): compile failed: error executing command
  (cd [home]/.cache/bazel/_bazel_jgrafton/75e5737b0f848bf1b7c3ab3303c8bf72/execroot/__main__ && \
  exec env - \
    GOARCH=amd64 \
    GOOS=linux \
    GOROOT=[home].cache/bazel/_bazel_jgrafton/75e5737b0f848bf1b7c3ab3303c8bf72/external/go1_7_6_linux_amd64 \
  bazel-out/host/bin/external/io_bazel_rules_go/go/tools/builders/compile external/go1_7_6_linux_amd64/bin/go -src pkg/credentialprovider/aws/aws_credentials.go -src pkg/credentialprovider/aws/aws_credentials_test.go -dep k8s.io/kubernetes/pkg/credentialprovider -dep github.com/aws/aws-sdk-go/aws -dep github.com/aws/aws-sdk-go/service/ecr -dep k8s.io/kubernetes/pkg/credentialprovider -dep github.com/aws/aws-sdk-go/aws -dep github.com/aws/aws-sdk-go/aws/request -dep github.com/aws/aws-sdk-go/aws/session -dep github.com/aws/aws-sdk-go/service/ecr -dep github.com/golang/glog -o 'bazel-out/local-fastbuild/bin/pkg/credentialprovider/aws/~lib~/go_default_test.o' -trimpath . -I . -I 'bazel-out/local-fastbuild/bin/pkg/credentialprovider/~lib~/' -I 'bazel-out/local-fastbuild/bin/vendor/github.com/aws/aws-sdk-go/aws/~lib~/' -I 'bazel-out/local-fastbuild/bin/vendor/github.com/aws/aws-sdk-go/service/ecr/~lib~/' -I 'bazel-out/local-fastbuild/bin/pkg/credentialprovider/~lib~/' -I 'bazel-out/local-fastbuild/bin/vendor/github.com/aws/aws-sdk-go/aws/~lib~/' -I'bazel-out/local-fastbuild/bin/vendor/github.com/aws/aws-sdk-go/aws/request/~lib~/' -I 'bazel-out/local-fastbuild/bin/vendor/github.com/aws/aws-sdk-go/aws/session/~lib~/' -I 'bazel-out/local-fastbuild/bin/vendor/github.com/aws/aws-sdk-go/service/ecr/~lib~/' -I 'bazel-out/local-fastbuild/bin/vendor/github.com/golang/glog/~lib~/' --)
                                                  
Use --sandbox_debug to see verbose messages from the sandbox
panic: runtime error: index out of range          
                                                  
goroutine 1 [running]:                            
panic(0x8b82c0, 0xc4200100d0)                     
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
cmd/compile/internal/gc.(*importer).pkg(0xc420474000, 0x38)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:216 +0xb1
cmd/compile/internal/gc.(*importer).fieldName(0xc420474000, 0x0)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:603 +0x397
cmd/compile/internal/gc.(*importer).field(0xc420474000, 0x5)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:545 +0x3d
cmd/compile/internal/gc.(*importer).fieldList(0xc420474000, 0xffffffffffffff16, 0xc420999ea0, 0x90)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:536 +0x8f
cmd/compile/internal/gc.(*importer).typ(0xc420474000, 0xc420999d60)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:483 +0xc77
cmd/compile/internal/gc.(*importer).typ(0xc420474000, 0xffffffffffffff12)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:407 +0x148
cmd/compile/internal/gc.(*importer).typ(0xc420474000, 0xffffffffffffff14)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:487 +0xfde
cmd/compile/internal/gc.(*importer).typ(0xc420474000, 0xc4208e09a0)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:474 +0xbf4
cmd/compile/internal/gc.(*importer).field(0xc420474000, 0xc420a56870)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:546 +0x55
cmd/compile/internal/gc.(*importer).fieldList(0xc420474000, 0xffffffffffffff16, 0xc420999810, 0x90)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:536 +0x8f
cmd/compile/internal/gc.(*importer).typ(0xc420474000, 0xc4209995e0)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:483 +0xc77
cmd/compile/internal/gc.(*importer).typ(0xc420474000, 0x0)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:407 +0x148
cmd/compile/internal/gc.(*importer).obj(0xc420474000, 0xfffffffffffffffd)
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:282 +0x967
cmd/compile/internal/gc.Import(0xc420060180)      
        /usr/local/go/src/cmd/compile/internal/gc/bimport.go:97 +0x3b2
cmd/compile/internal/gc.importfile(0xc420410800, 0x0, 0x0, 0x0)
        /usr/local/go/src/cmd/compile/internal/gc/main.go:843 +0x9b1
cmd/compile/internal/gc.(*parser).importdcl(0xc4204107e0)
        /usr/local/go/src/cmd/compile/internal/gc/parser.go:341 +0x178
cmd/compile/internal/gc.(*parser).import_(0xc4204107e0)
        /usr/local/go/src/cmd/compile/internal/gc/parser.go:300 +0x71
cmd/compile/internal/gc.(*parser).file(0xc4204107e0)
        /usr/local/go/src/cmd/compile/internal/gc/parser.go:268 +0x5e
cmd/compile/internal/gc.parse_file(0xc420060240)  
        /usr/local/go/src/cmd/compile/internal/gc/parser.go:31 +0x54
cmd/compile/internal/gc.Main()                    
        /usr/local/go/src/cmd/compile/internal/gc/main.go:334 +0x14f4
cmd/compile/internal/amd64.Main()                 
        /usr/local/go/src/cmd/compile/internal/amd64/galign.go:93 +0x2fa
main.main()                                       
        /usr/local/go/src/cmd/compile/main.go:33 +0x2a3
```

--- a/go/private/actions/compile.bzl
+++ b/go/private/actions/compile.bzl
@@ -20,7 +20,7 @@ load("@io_bazel_rules_go//go/private:providers.bzl",
     "get_searchpath",
 )
 
-def emit_compile(ctx, go_toolchain, sources, golibs, mode, out_object, gc_goopts):
+def emit_compile(ctx, go_toolchain, sources, golibs, mode, out_lib, gc_goopts):
   """Construct the command line for compiling Go code.
 
   Args:
@@ -29,7 +29,7 @@ def emit_compile(ctx, go_toolchain, sources, golibs, mode, out_object, gc_goopts
     golibs: a depset of representing all imported libraries.
     mode: Controls the compilation setup affecting things like enabling profilers and sanitizers.
       This must be one of the values in common.bzl#compile_modes
-    out_object: the object file that should be produced
+    out_lib: the archive file that should be produced
     gc_goopts: additional flags to pass to the compiler.
   """
 
@@ -48,11 +48,11 @@ def emit_compile(ctx, go_toolchain, sources, golibs, mode, out_object, gc_goopts
     inputs += [get_library(golib, mode)]
     args += ["-dep", golib.importpath]
     args += ["-I", get_searchpath(golib,mode)]
-  args += ["-o", out_object.path, "-trimpath", ".", "-I", "."]
+  args += ["-o", out_lib.path, "-trimpath", ".", "-I", "."]
   args += ["--"] + gc_goopts + go_toolchain.flags.compile + cgo_sources
   ctx.action(
       inputs = list(inputs),
-      outputs = [out_object],
+      outputs = [out_lib],
       mnemonic = "GoCompile",
       executable = go_toolchain.tools.compile,
       arguments = args,

--- a/go/private/actions/library.bzl
+++ b/go/private/actions/library.bzl
@@ -78,20 +78,36 @@ def emit_library(ctx, go_toolchain, srcs, deps, cgo_object, library, want_covera
   lib_name = importpath + ".a"
   mode_fields = {} # These are added to the GoLibrary provider directly
   for mode in compile_modes:
-    out_lib = ctx.new_file("~{}~{}~/{}".format(mode, ctx.label.name, lib_name))
-    out_object = ctx.new_file("~{}~{}~/{}.o".format(mode, ctx.label.name, importpath))
+    out_dir = "~{}~{}~".format(mode, ctx.label.name)
+    out_lib = ctx.new_file("{}/{}".format(out_dir, lib_name))
     searchpath = out_lib.path[:-len(lib_name)]
     mode_fields[library_attr(mode)] = out_lib
     mode_fields[searchpath_attr(mode)] = searchpath
-    go_toolchain.actions.compile(ctx,
-        go_toolchain = go_toolchain,
-        sources = go_srcs,
-        golibs = direct,
-        mode = mode,
-        out_object = out_object,
-        gc_goopts = gc_goopts,
-    )
-    go_toolchain.actions.pack(ctx, go_toolchain, out_lib, [out_object] + extra_objects)
+    if len(extra_objects) == 0:
+      go_toolchain.actions.compile(ctx,
+          go_toolchain = go_toolchain,
+          sources = go_srcs,
+          golibs = direct,
+          mode = mode,
+          out_lib = out_lib,
+          gc_goopts = gc_goopts,
+      )
+    else:
+      partial_lib = ctx.new_file("{}/~partial.a".format(out_dir))
+      go_toolchain.actions.compile(ctx,
+          go_toolchain = go_toolchain,
+          sources = go_srcs,
+          golibs = direct,
+          mode = mode,
+          out_lib = partial_lib,
+          gc_goopts = gc_goopts,
+      )
+      go_toolchain.actions.pack(ctx,
+          go_toolchain = go_toolchain,
+          in_lib = partial_lib,
+          out_lib = out_lib,
+          objects = extra_objects,
+      )
 
   dylibs = []
   if cgo_object:

--- a/go/private/actions/pack.bzl
+++ b/go/private/actions/pack.bzl
@@ -12,20 +12,25 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-def emit_pack(ctx, go_toolchain, out_lib, objects):
+def emit_pack(ctx, go_toolchain, in_lib, out_lib, objects):
   """Construct the command line for packing objects together.
 
   Args:
     ctx: The skylark Context.
+    in_lib: the archive that should be copied and appended to.
     out_lib: the archive that should be produced
     objects: an iterable of object files to be added to the output archive file.
   """
+  arguments = [
+      go_toolchain.tools.go.path,
+      in_lib.path,
+      out_lib.path,
+  ] + [obj.path for obj in objects]
   ctx.action(
-      inputs = objects + go_toolchain.data.tools,
+      inputs = [in_lib] + objects + go_toolchain.data.tools,
       outputs = [out_lib],
       mnemonic = "GoPack",
-      executable = go_toolchain.tools.go,
-      arguments = ["tool", "pack", "c", out_lib.path] + [a.path for a in objects],
+      executable = go_toolchain.tools.pack,
+      arguments = arguments,
       env = go_toolchain.env,
   )
-

--- a/go/private/go_toolchain.bzl
+++ b/go/private/go_toolchain.bzl
@@ -39,27 +39,28 @@ def _go_toolchain_impl(ctx):
           pack = emit_pack,
       ),
       paths = struct(
-        root = ctx.attr.root,
+          root = ctx.attr.root,
       ),
       tools = struct(
-        go = ctx.executable.go,
-        asm = ctx.executable._asm,
-        compile = ctx.executable._compile,
-        link = ctx.executable._link,
-        cgo = ctx.executable._cgo,
-        test_generator = ctx.executable._test_generator,
-        extract_package = ctx.executable._extract_package,
+          go = ctx.executable.go,
+          asm = ctx.executable._asm,
+          compile = ctx.executable._compile,
+          pack = ctx.executable._pack,
+          link = ctx.executable._link,
+          cgo = ctx.executable._cgo,
+          test_generator = ctx.executable._test_generator,
+          extract_package = ctx.executable._extract_package,
       ),
       flags = struct(
-        compile = ctx.attr._go_toolchain_flags.compile_flags,
-        link = ctx.attr.link_flags,
-        link_cgo = ctx.attr.cgo_link_flags,
+          compile = ctx.attr._go_toolchain_flags.compile_flags,
+          link = ctx.attr.link_flags,
+          link_cgo = ctx.attr.cgo_link_flags,
       ),
       data = struct(
-        tools = ctx.files.tools,
-        stdlib = ctx.files.stdlib,
-        headers = ctx.attr.headers,
-        crosstool = ctx.files._crosstool,
+          tools = ctx.files.tools,
+          stdlib = ctx.files.stdlib,
+          headers = ctx.attr.headers,
+          crosstool = ctx.files._crosstool,
       ),
       external_linker = ctx.attr._external_linker,
   )]
@@ -78,6 +79,11 @@ def _compile(bootstrap):
   if bootstrap:
     return None
   return Label("//go/tools/builders:compile")
+
+def _pack(bootstrap):
+  if bootstrap:
+    return None
+  return Label("//go/tools/builders:pack")
 
 def _link(bootstrap):
   if bootstrap:
@@ -116,6 +122,7 @@ go_toolchain = rule(
         # Tools, missing from bootstrap toolchains
         "_asm": attr.label(allow_files = True, single_file = True, executable = True, cfg = "host", default = _asm),
         "_compile": attr.label(allow_files = True, single_file = True, executable = True, cfg = "host", default = _compile),
+        "_pack": attr.label(allow_files = True, single_file = True, executable = True, cfg = "host", default = _pack),
         "_link": attr.label(allow_files = True, single_file = True, executable = True, cfg = "host", default = _link),
         "_cgo": attr.label(allow_files = True, single_file = True, executable = True, cfg = "host", default = _cgo),
         "_test_generator": attr.label(allow_files = True, single_file = True, executable = True, cfg = "host", default = _test_generator),

--- a/go/tools/builders/BUILD.bazel
+++ b/go/tools/builders/BUILD.bazel
@@ -71,3 +71,9 @@ go_tool_binary(
     ],
     visibility = ["//visibility:public"],
 )
+
+go_tool_binary(
+    name = "pack",
+    srcs = ["pack.go"],
+    visibility = ["//visibility:public"],
+)

--- a/go/tools/builders/compile.go
+++ b/go/tools/builders/compile.go
@@ -79,7 +79,7 @@ func run(args []string) error {
 	for _, path := range search {
 		goargs = append(goargs, "-I", abs(path))
 	}
-	goargs = append(goargs, "-o", *output)
+	goargs = append(goargs, "-pack", "-o", *output)
 	goargs = append(goargs, flags.Args()...)
 	goargs = append(goargs, sources...)
 	cmd := exec.Command(gotool, goargs...)

--- a/go/tools/builders/pack.go
+++ b/go/tools/builders/pack.go
@@ -1,0 +1,63 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// pack copies an .a file and appends a list of .o files to the copy using
+// go tool pack. It is invoked by the Go rules as an action.
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+)
+
+func run(args []string) error {
+	if len(args) < 4 {
+		return fmt.Errorf("Usage: pack gotool in.a out.a obj.o...")
+	}
+	gotool := args[0]
+	inArchive := args[1]
+	outArchive := args[2]
+	objects := args[3:]
+
+	if err := copyFile(inArchive, outArchive); err != nil {
+		return err
+	}
+	packArgs := append([]string{"tool", "pack", "r", outArchive}, objects...)
+	cmd := exec.Command(gotool, packArgs...)
+	return cmd.Run()
+}
+
+func main() {
+	if err := run(os.Args[1:]); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func copyFile(inPath, outPath string) error {
+	inFile, err := os.Open(inPath)
+	if err != nil {
+		return err
+	}
+	defer inFile.Close()
+	outFile, err := os.Create(outPath)
+	if err != nil {
+		return err
+	}
+	defer outFile.Close()
+	_, err = io.Copy(outFile, inFile)
+	return err
+}

